src/pretix/base/models/event.py

```diff
@@ -544,6 +546,59 @@ def get_users_with_permission(self, permission):
             Q(is_superuser=True) | Q(twp=True)
         )
 
+    @staticmethod
+    def has_paid_things(event):
+        item = apps.get_model('pretixbase', 'Item')
+        item_variation = apps.get_model('pretixbase', 'ItemVariation')
+        return item.objects.filter(event=event, default_price__gt=0).exists()\
+            or item_variation.objects.filter(item__event=event, default_price__gt=0).exists()
+
+    @staticmethod
+    def has_payment_provider(event):
+        result = False
+        for provider in event.get_payment_providers().values():
+            if provider.is_enabled and provider.identifier != 'free':
+                result = True
+                break
+        return result
+
+    @staticmethod
+    def clean_payment_methods(event):
+        if Event.has_paid_things(event) and not Event.has_payment_provider(event):
+            raise ValidationError(_('You have configured at least one paid product but have not enabled any payment '
+                                    'methods.'))
+
+    @staticmethod
+    def clean_quotas(event):
+        if event is None or not event.quotas.exists():
+                raise ValidationError(_('You need to configure at least one quota to sell anything.'))
+
+    @staticmethod
+    def clean_live(event):
+        Event.clean_payment_methods(event)
+        Event.clean_quotas(event)
+
+    @staticmethod
+    def clean_slug(organizer, event, slug):
```

Comment:
Just to make the distinction clear: This and the following two methods are rightfully static methods, since they do not depend on an ``Event`` object at all and can be used on serializer or form level even before the ``Event`` object is constructed!